library/farmosUtil/farmosUtil.transaction.js

```diff
@@ -28,6 +28,71 @@
  * If an operation was never attempted (and thus also not undone) the attribute for that operation will be `undefined`.
  *
  * @category Utilities
+ * 
+ * Executes a transaction with the provided operations.
+ *
+ * @param {Array<Object>} operations the operations to execute as a transaction.
+ * Each operation must have the following structure:
+ *
+ * ```
+ * {
+ * name: string,
+ * do: (Object) => async function performs an action and returns a result (e.g. creates a log, asset or quantity.) The argument has the same format as the return value. It contains the result of all prior operations, allowing them to be used by future operations.
+ * undo: (Object) => async function that undoes the action performed by `do` (e.g. deletes a log, asset or quantity). The argument has the same format as the return value and contains the result of all successfully completed operations.
+ * }
+ * ```
+ *
+ * @example
+ * // Create a transaction with a plant asset, a quantity, and a log
+ * const operations = [
+ *   {
+ *     name: 'createPlantAsset',
+ *     do: async () => {
+ *       return await farmosUtil.createPlantAsset(
+ *         '2023-10-01', // Example date
+ *         'ZUCCHINI',   // Example crop name
+ *         'Planting zucchini in the greenhouse' // Example comment
+ *       );
+ *     },
+ *     undo: async (results) => {
+ *       await farmosUtil.deletePlantAsset(results['createPlantAsset'].id);
+ *     },
+ *   },
+ *   {
+ *     name: 'createQuantity',
+ *     do: async () => {
+ *       return await farmosUtil.createStandardQuantity(
+ *         'count',
+ *         50, // Example quantity value
+ *         'Harvested Zucchini', // Example description
+ *         'KG' // Example unit
+ *       );
+ *     },
+ *     undo: async (results) => {
+ *       await farmosUtil.deleteStandardQuantity(results['createQuantity'].id);
```

Comment:
We have been working on a change in the way we `undo` quantities -  so the following would not have been in any of examples you've looked at yet. But we should include it here.

The delete of the quantity should be within with an `if` that checks if it needs to be deleted before attempting to delete it.  This check is helpful because when a log that references a quantity is deleted, the quantity is also deleted. So, if for example the seeding log created below were deleted by it's `undo`, then this quantity would have already been deleted.  Thus, attempting to delete it in its `undo` would fail.  So we can check first with:
```
if (results['createLog'] != 'undone') {
  // put the delete operation here.
}
```